### PR TITLE
Add limit pushdown to Information Schema Connector

### PR DESCRIPTION
**Pull request by linxingyuan1102**
#998

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaPageSourceProvider.java [hunk 1]
```diff
@@ -110,42 +111,43 @@ private InternalTable getInternalTable(ConnectorSession connectorSession, Connec
...
         }
 
         throw new IllegalArgumentException(format("table does not exist: %s", table));
     }
 
-    private InternalTable buildColumns(Session session, Set<QualifiedTablePrefix> prefixes)
+    private InternalTable buildColumns(Session session, Set<QualifiedTablePrefix> prefixes, OptionalLong limit)
     {
         InternalTable.Builder table = InternalTable.builder(informationSchemaTableColumns(TABLE_COLUMNS));
+        long recordCnt = 0;
```
Comment (electrum) [inline presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaPageSourceProvider.java]: Nit: we don't abbreviate in our code style. Please name this `recordCount` or just `count`.

File: presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaPageSourceProvider.java [hunk 2]
```diff
@@ -167,15 +169,20 @@ private InternalTable buildColumns(Session session, Set<QualifiedTablePrefix> pr
                             column.getExtraInfo(),
                             column.getComment());
                     ordinalPosition++;
+                    recordCnt++;
+                    if (limit.isPresent() && recordCnt == limit.getAsLong()) {
```
Comment (electrum) [inline presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaPageSourceProvider.java]: I think we can simplify this by having `InternalTable.Builder` keep track of the size. Then we can add a method that takes `OptionalLong` to do the check:
```
if (table.atLimit(limit)) {
    return table.build();
}

**Comment by linxingyuan1102** [comment]
@electrum  Thank you for the review. I've updated the PR.

**Comment by electrum** [comment]
Sorry for the delay on this. I restarted the failed Travis job and will merge after it passes.

**Pull request merged**